### DISCUSSION TITLE
Increase minimum fees on Fetch-ai Network Mainnet

### DISCUSSION THREAD

**Discussion by rgvnFET** [fetchai/FIPs#8] [Ideas]
Many validators on the Fetch-ai Network do not have minimum gas fees set to their node for accepting transactions. This can easily lead to network congestion from spam attacks since you can spam transactions at no cost. To mitigate this problem, the Fetch-ai Network team believes it be necessary for as many validators as possible (hopefully all) to set their minimum-gas-prices value to at least 2 aFET on their validator node.

Other networks in the ecosystem such as [Cosmos](https://github.com/cosmos/cosmos-sdk/discussions/8224#discussioncomment-457169) and [Osmosis](https://commonwealth.im/osmosis/discussion/6298-draft-minimum-gas-fee) have already had their validators and community discuss this and came to the same conclusion that our team has, which is that this is the safest and easiest way to prevent these network attacks.

The Fetch-ai Network team have already done testing with having minimum fees on both the previous Capricorn testnet and even now with the Dorado testnet and we experienced no issues from this.

We'd like to gauge the community and validator's sentiment on this idea, so if you have any suggestions then feel free to post below.

**Comment by qf3l3k** [comment]
no fees might potentially lead to automated tx spams, so definitely good idea. proposed amount 2aFET also sounds reasonable considering denom FET token has.

**Comment by zdeadex** [comment]
it's better to prevent so I would recommend to increase the minimum fees on the mainnet chain!

**Reply by DDeAlmeida** [reply]
Definitely pushing for this

**Comment by JulianPscheid** [comment]
2 aFET sounds like a good starting point. This has the support from STAKING.LAND.

**Comment by packetstracer** [comment]
we support the proposal to avoid spamming the network, but having in mind aFET denom specs seems like 2 aFET is very small. With just 1$ you could send tons of transactions. 2 aFET is better than nothing but not very safe IMO.

**Reply by Atari-buzzk1LL** [reply]
The idea here is to start small, but absolutely work our way up to having higher fees than that. Considering we're trying to get validators on board who are taking on transactions that are free, we figured making them just barely more than free would edge them into it.

**Comment by Teasel-Ian** [comment]
The proposed gas price of 2afet is unlikely to deter spamming.  A budget of, say, 10FET, would buy 5x10<sup>18</sup> gas.  At around  60,000 afet for a send tx, and 150,000 afet for a staking tx, an attacker could run 20 trillion of each and still have change from a dollar!

As Atari-buzzk1LL says, we'll almost certainly need to increase gas prices in future.  Not to anywhere near ETH / BTC levels, where network capacity is woefully lacking, but perhaps to a level where it's economically unattractive to flood the network beyond its available throughput.  Assuming proposal 8 passes, our network capacity will be around 30M gas / minute, and the gas price will ultimately need to reflect that limit.

From a technical viewpoint, however, there's a world of difference betweeen a zero and a non-zero gas price.  Even with a trivial 2afet minimum gas price, many currently-used API calls, and fetchd commands, will flat-out fail unless additional parameters are supplied.  Also, shortcomings in the current Cosmos SDK's gas estimation mean that developers will likely need to experiment with gas-adjustment values in order to get their code working reliably.  That's one reason that the dorado testnet has imposed a minimum gas price (of 1000000000atestfet) for some time now.  So, IMHO, the main benefit of this proposal is to ensure that developers support non-zero gas prices in their code.  Once all applications running on the network expect to pay for gas, the actual gas price can safely fluctuate according to the demand / threat landscape at that point in time.

**Reply by markos** [reply]
completely agree, either do this properly or not at all. For experimenting with non-zero gas prices, imho, that's what the testnets are for.

**Reply by packetstracer** [reply]
agree, as previously stated

**Comment by bartcrpt** [comment]
agree, slowly increasing of gas fee is the normal way to improve network

**Reply by Teasel-Ian** [reply]
I understand that Fetch.ai experts, with far better mathematical skills than I, have done a rigorous analysis of gas prices (I'm just the SRE guy, so take the following with a big pinch of salt, and watch this space!)... but to give casual readers a vague ballpark indication of where we're ultimately headed, once all the exciting AEA/Colearn developments land on mainnet and go mainstream, here are a couple of graphs representing gas prices from the point of view of both attackers and legitimate users... 
![image](https://user-images.githubusercontent.com/83638255/181390659-393a9cf7-a035-4ee5-b268-ae591cda81c1.png)![image](https://user-images.githubusercontent.com/83638255/181390692-e9639d59-57af-470b-a764-1b2e10722d12.png)
The interesting "end game" region is somewhere in the centre of these graphs. 
So, for example, a gas price of 10,000,000,000 afet would mean legitimate users could run 2000 simple transactions for a dollar, and attackers would need to spend $400 per day to flood the network.
A higher gas price of 100,000,000,000 afet would increase the price of a simple transaction to $0.005, but would cost hackers $4000 per day to cause sustained annoyance.
Bear in mind the many caveats glossed over in the above.  AEA/Colearn transactions are likely to use far more gas than a simple "tx bank send",  USD/FET will (hopefully!) rise by at least an order of magnitude once the "crypto winter" thaws, etc etc.  But this approximation nevertheless gives a vague idea of the rough "number of zeroes" we're headed towards in the Fetch.ai end game, to avoid both the threat of DoS attack, and also the horrendous $2 (or even $60) transaction fees seen on other blockchain networks.

A non-zero, 2afet, gas price will force developers to support gas in their code, which is a Good Thing to enforce in the short term (I myself had alarms suddenly going off all over the place, the day dorado introduced gas fees, and had to make some swift code changes!).  But in the medium term we're going to need to gradually add at least 9-11 more zeroes.

**Comment by kiranezu** [comment]
At MOBIX.AI we think it makes a lot of sense to introduce FET transaction fees. However, It will take us some time to implement the support of fees in our system (which consists of several components that have to be aligned). Therefore, we ask to introduce the fees effectively at the end of September 2022. This would grant us sufficient time for integrating and fully testing the transaction fee support within our products.

**Reply by Atari-buzzk1LL** [reply]
What we'd like to see happen is for most validators to set their nodes to only accept transaction fees, but we understand that since we can't make anybody do anything there is a good chance that a handful of validators will still be leaving their nodes at accepting 0 fee transactions.  This will provide the MOBIX team plenty of time to be able to do the necessary testing while also giving the Fetch-ai Network team the ability to begin the stages of securing the network better. As outlined by @ianharris-fetch in his above comments, this will need to happen in many stages and could actually take quite a while to get to a point that's favourable for security, so we'd like to at least start getting it out there that this is something we're working towards.

**Comment by Atari-buzzk1LL** [comment]
I just wanted to let the community know that we will be leaving this purely as a discussion and not moving onto a Fetch Improvement Proposal (FIP) for 2-3 weeks maximum to allow the MOBIX team the time that has requested by @kiranezu for having an implementation ready for the proposal. I wanted to inform the community about this waiting period so there is no confusion on why the FIP is not being put out.

**Comment by Teasel-Github** [comment]
Update from the MOBIX team... https://mobix.ai/2022/09/13/mobix-app-update-september-2022/
